File: src/main/java/org/snake/views/BlockView.java
package org.snake.views;

import org.snake.models.Block;

import java.awt.*;

import static com.google.common.base.Preconditions.checkNotNull;

public class BlockView implements FigureView {

	private Block model;

	public BlockView(Block model) {
		this.model = checkNotNull(model);
	}

	@Override
	public boolean isEmptySpace() {
		return model.isEmptySpace();
	}

	@Override
	public void draw(Graphics canvas, int x, int y){
		canvas.setColor(Color.lightGray);
		canvas.fill3DRect(0, 0, x, y, true);	
	}

}

File: src/main/kotlin/org/snake/views/BlockView.kt
package org.snake.views

import org.snake.models.Block
import java.awt.Graphics

class BlockView(val model: Block) : FigureView {
    override val isEmptySpace: Boolean
        get() = model.isEmptySpace

    override fun draw(canvas: Graphics, x: Int, y: Int, width: Int, height: Int) {
        canvas.color = BLOCK_COLOR
        canvas.fill3DRect(x, y, width, height, true)
    }
}